app/javascript/vm-snapshot-form/register.jsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { SnapshotForm } from '@manageiq/react-ui-components/dist/vm-snapshot-form';
+
+ManageIQ.component.newRegistry.define('SnapshotForm', ManageIQ.component.reactBlueprint(props => (
+  <SnapshotForm {...props} />
+)));
+
+const mountVmSnapshotForm = (createUrl, cancelUrl, errorMessages, labels, vendor) => {
+  ManageIQ.component.newInstance(
+    'SnapshotForm',
+    {
+      onSubmit: (values) => {
+        if (values.snap_memory) {
+          values.snap_memory = 1;
+        }
+        window.miqAjaxButton(createUrl, values);
+      },
+      onCancel: () => {
+        window.miqAjaxButton(cancelUrl);
+      },
+      hideName: vendor === 'redhat',
+      nameRequired: vendor === 'redhat',
+      descriptionRequired: vendor === 'redhat',
+      errorMessages,
+      labels,
+    },
+    $('#snap-form').get(0),
+  );
+};
+
+$(() => {
```

Comment:
TBH I don't understand why we are solving the problem this way.

IMHO the right way to do this is:
1. expose the top level / high level react component so its available to the registry
2. use a rails view helper to mount the component on the dom as demonstrated at https://github.com/priley86/miq_v2v_ui_plugin/blob/master/app/views/migration/index.html.haml#L2
this will allow you to pass the right data in the place you would expect react to be rendering (Rather relaying on jquery functions).
You would be able to pass any ruby props at that stage as well.